pkg/query-service/app/logparsingpipeline/pipelineBuilder.go

```diff
@@ -47,13 +62,28 @@ func PreparePipelineProcessor(pipelines []model.Pipeline) (map[string]interface{
 		processor := model.Processor{
 			Operators: v.Config,
 		}
-		name := constants.LogsPPLPfx + v.Alias
+		name := CollectorConfProcessorName(v)
 		processors[name] = processor
 		names = append(names, name)
 	}
 	return processors, names, nil
 }
 
+func PipelineFilterExpr(serializedFilter string) (string, error) {
+	var filterset v3.FilterSet
+	err := json.Unmarshal([]byte(serializedFilter), &filterset)
+	if err != nil {
+		return "", errors.Wrap(err, "could not unmarshal pipeline filterset json")
+	}
```

Comment:
Is there a reason to send the JSON String and unmarshal here rather than sending the filter data structure in the payload directly?